types/hubot/index.d.ts

```diff
@@ -86,31 +93,24 @@ export class Message {
     constructor(user: User, done?: boolean);
     id: string;
     user: User;
-    text: string | null;
     room: string;
+    done: boolean;
+    text?: string;
     finish(): void;
 }
 
 export class TextMessage extends Message {
-    text: string;
-
     constructor(user: User, text: string, id: string);
 
     match(regex: RegExp): RegExpMatchArray;
     toString(): string;
 }
 
-export class EnterMessage extends Message {
-    text: null;
-}
+export class EnterMessage extends Message {}
 
-export class LeaveMessage extends Message {
-    text: null;
-}
+export class LeaveMessage extends Message {}
 
-export class TopicMessage extends TextMessage {
-    text: string;
-}
+export class TopicMessage extends TextMessage {}
```

Comment:
FWIW these classes will all be treated as "the same" by the type system because they have no differing props. It's a problem with the old types, though, so it doesn't really matter, but a "fix" for this is to add in a fake private thing like `private _;` just to make them all different.

Reply:
I didn't know that, let me fix that!